src/gsy_e/models/market/__init__.py

```diff
@@ -271,3 +291,42 @@ def total_earned(self, seller: str) -> float:
         """Return the aggregated money earned by the passed-in seller."""
 
         return sum(trade.trade_price for trade in self.trades if trade.seller == seller)
+
+    @staticmethod
+    @abstractmethod
```

Comment:
Hmm, I am generally not a fan of abstractmethods with implementations, however for this case it is okay to have. The alternative would be to implement it in the OneSidedMarket class however it does not make that much of a difference here.

Reply:
Actually, having this as abstractmethod, I need to implement it in the OneSidedMarket and all subclasses, right?
let me remove the abstractmethod here and let the subclasses re-use it (as intended)

Reply:
[575a5ba](https://github.com/gridsingularity/gsy-e/pull/1565/commits/575a5ba109398e065ca763f6ea44371fd8e66bda)